Week2/homeworkWeek2.js

```diff
@@ -0,0 +1,125 @@
+
+async function queryDatabase() {
+  help();
+  prompt.start();
+  try {
+    const use_database = `use world`;
+    await execQuery(use_database);
+
+    const options = await input(['number']);
+    const inputNumber = options.number;
+
+    switch (inputNumber) {
+      case '1':
+        const countries = await input(['country']);
+        const input_country = countries.country;
+        const firstQuery = `SELECT city.name FROM city JOIN country ON country.capital = city.id WHERE country.name = ?;`;
+        const firstResults = await execQuery(firstQuery, input_country);
+        for (result of firstResults) {
+          console.log(`The capital of ${input_country.toUpperCase()} is "${result.name}"`);
+        }
+        break;
+      case '2':
+        const regions = await input(['region']);
+        const input_region = regions.region;
+        const secondQuery = `SELECT DISTINCT countrylanguage.language AS 'LanguagesSpokenPerRegion' FROM countrylanguage 
+                              JOIN country ON country.code = countrylanguage.countrycode
+                              WHERE country.region = ?;
+                              `;
+        const secondResults = await execQuery(secondQuery, input_region);
+        console.log(`The languages spoken of ${input_region.toUpperCase()} are:`);
+        for (result of secondResults) {
+          console.log(`${JSON.stringify(result.LanguagesSpokenPerRegion)}`);
+        }
+        break;
+      case '3':
+        const languages = await input(['language']);
+        const input_language = languages.language;
+        const thirdQuery = `SELECT COUNT(city.name) AS 'NumberOfCitiesPerLanguage' FROM city 
+                             JOIN countrylanguage ON city.countrycode = countrylanguage.countrycode
+                             WHERE countrylanguage.language = ?;
+                             `;
+        const thirdResults = await execQuery(thirdQuery, input_language);
+        console.log(`The number of cities in which ${input_language.toUpperCase()} is spoken is:`);
+        for (result of thirdResults) {
+          console.log(result.NumberOfCitiesPerLanguage);
+        }
+        break;
+      case '4':
+        const regionAndLanguage = await input(['region', 'language']);
+        const inputRegion = regionAndLanguage.region;
+        const inputLanguage = regionAndLanguage.language;
+        const fourthQuery = `SELECT country.name FROM country 
+                              JOIN countrylanguage ON country.code = countrylanguage.countrycode
+                              WHERE countrylanguage.isofficial = 'T' AND country.region = ? AND countrylanguage.language = ?;
+                              `;
+        const fourthResults = await execQuery(fourthQuery, [inputRegion, inputLanguage]);
+        console.log(
+          `The countries in which ${inputLanguage.toUpperCase()} is spoken in the ${inputRegion.toUpperCase()} are:`,
+        );
+        if (fourthResults.length === 0) {
```

Comment:
I like that you handled it in the JavaScript part. One could write a stored procedure in SQL for this. But this is more elegant.